[2024-06-03T07:37:41.372+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-03T07:37:41.390+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: my_first_dag.tsk_extract_from_api manual__2024-06-03T07:37:18.982265+00:00 [queued]>
[2024-06-03T07:37:41.395+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: my_first_dag.tsk_extract_from_api manual__2024-06-03T07:37:18.982265+00:00 [queued]>
[2024-06-03T07:37:41.396+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 3
[2024-06-03T07:37:41.411+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): tsk_extract_from_api> on 2024-06-03 07:37:18.982265+00:00
[2024-06-03T07:37:41.415+0000] {standard_task_runner.py:63} INFO - Started process 4859 to run task
[2024-06-03T07:37:41.418+0000] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'my_first_dag', 'tsk_extract_from_api', 'manual__2024-06-03T07:37:18.982265+00:00', '--job-id', '18', '--raw', '--subdir', 'DAGS_FOLDER/myfirstdag.py', '--cfg-path', '/tmp/tmpowyze7_p']
[2024-06-03T07:37:41.420+0000] {standard_task_runner.py:91} INFO - Job 18: Subtask tsk_extract_from_api
[2024-06-03T07:37:41.457+0000] {task_command.py:426} INFO - Running <TaskInstance: my_first_dag.tsk_extract_from_api manual__2024-06-03T07:37:18.982265+00:00 [running]> on host ip-172-31-35-145.ap-southeast-1.compute.internal
[2024-06-03T07:37:41.519+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='airflow@example.com' AIRFLOW_CTX_DAG_OWNER='jeqiu' AIRFLOW_CTX_DAG_ID='my_first_dag' AIRFLOW_CTX_TASK_ID='tsk_extract_from_api' AIRFLOW_CTX_EXECUTION_DATE='2024-06-03T07:37:18.982265+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-03T07:37:18.982265+00:00'
[2024-06-03T07:37:41.520+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-03T07:37:41.533+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-03T07:37:41.533+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/ubuntu/e2e_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/e2e_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/e2e_venv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/e2e_venv/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/e2e_venv/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/airflow/dags/myfirstdag.py", line 32, in extract_from_api
    dt_string = kwargs['date_string']
                ~~~~~~^^^^^^^^^^^^^^^
KeyError: 'date_string'
[2024-06-03T07:37:41.538+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=my_first_dag, task_id=tsk_extract_from_api, run_id=manual__2024-06-03T07:37:18.982265+00:00, execution_date=20240603T073718, start_date=20240603T073741, end_date=20240603T073741
[2024-06-03T07:37:41.553+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 18 for task tsk_extract_from_api ('date_string'; 4859)
[2024-06-03T07:37:41.590+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-03T07:37:41.602+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-03T07:37:41.602+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
